python/tk_desktop/desktop_engine_site_implementation.py

```diff
@@ -342,6 +342,38 @@ def run(self, splash, version, **kwargs):
         :param startup_version: Version of the Desktop Startup code. Can be omitted.
         :param startup_descriptor: Descriptor of the Desktop Startup code. Can be omitted.
         """
+        # Initialize Qt app
+        from tank.platform.qt import QtGui
+
+        app = QtGui.QApplication.instance()
+        if app is None:
+            app = QtGui.QApplication(sys.argv)
+
+        # update the app icon
+        icon = QtGui.QIcon(":tk-desktop/default_systray_icon")
+        app.setWindowIcon(icon)
+
+        if splash:
+            splash.set_message("Building UI")
+
+        # setup the global look and feel
+        self._engine._initialize_dark_look_and_feel()
+
+        # load custom font
+        QtGui.QFontDatabase.addApplicationFont(":/tk-desktop/fonts/OpenSans-Bold.ttf")
+        QtGui.QFontDatabase.addApplicationFont(":/tk-desktop/fonts/OpenSans-Regular.ttf")
+        QtGui.QFontDatabase.addApplicationFont(":/tk-desktop/fonts/OpenSans-CondLight.ttf")
+        QtGui.QFontDatabase.addApplicationFont(":/tk-desktop/fonts/OpenSans-Light.ttf")
+
+        # merge in app specific look and feel
+        css_file = os.path.join(self._engine.disk_location, "resources", "desktop_dark.css")
+        with open(css_file) as f:
+            css = app.styleSheet() + "\n\n" + f.read()
+        app.setStyleSheet(css)
+
+        from .console import Console
```

Comment:
Suggest comments here saying why the console is being instantiated here instead of when it is used below. Also a comment on why the import is deferred would be nice.